File: app/src/main/java/com/example/telemedicine_lab5/signaling/SignalingMessage.kt
package com.example.telemedicine_lab5.signaling

import org.json.JSONObject

data class SignalingMessage(
    val type: String,
    val sdp: String? = null,
    val candidate: String? = null,
    val sdpMid: String? = null,
    val sdpMLineIndex: Int? = null,
)

fun parseSignalingMessage(payload: JSONObject): SignalingMessage {
    val sdpMLineIndex = when (val raw = payload.opt("sdpMLineIndex")) {
        is Number -> raw.toInt()
        is String -> raw.toIntOrNull()
        else -> null
    }

    return SignalingMessage(
        type = payload.optString("type"),
        sdp = payload.optString("sdp").ifBlank { null },
        candidate = payload.optString("candidate").ifBlank { null },
        sdpMid = payload.optString("sdpMid").ifBlank { null },
        sdpMLineIndex = sdpMLineIndex,
    )
}

File: flutter_app/lib/src/models/signaling_message.dart
class SignalingMessage {
  const SignalingMessage({
    required this.type,
    this.sdp,
    this.candidate,
    this.sdpMid,
    this.sdpMLineIndex,
  });

  final String type;
  final String? sdp;
  final String? candidate;
  final String? sdpMid;
  final int? sdpMLineIndex;

  factory SignalingMessage.fromMap(Map<String, dynamic> map) {
    return SignalingMessage(
      type: map['type']?.toString() ?? '',
      sdp: map['sdp']?.toString(),
      candidate: map['candidate']?.toString(),
      sdpMid: map['sdpMid']?.toString(),
      sdpMLineIndex: _parseInt(map['sdpMLineIndex']),
    );
  }

  static int? _parseInt(dynamic value) {
    if (value == null) return null;
    if (value is int) return value;
    return int.tryParse(value.toString());
  }
}
